elide-core/src/main/java/com/yahoo/elide/core/filter/Operator.java

```diff
@@ -5,26 +5,168 @@
  */
 package com.yahoo.elide.core.filter;
 
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.core.PersistentResource;
 import com.yahoo.elide.core.exceptions.InvalidPredicateException;
+import com.yahoo.elide.utils.coerce.CoerceUtil;
+
 import lombok.Getter;
 import lombok.RequiredArgsConstructor;
 
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
 /**
  * Operator enum for predicates.
  */
 @RequiredArgsConstructor
-public enum Operator {
-    IN("in", true),
-    NOT("not", true),
-    PREFIX("prefix", true),
-    POSTFIX("postfix", true),
-    INFIX("infix", true),
-    ISNULL("isnull", false),
-    NOTNULL("notnull", false),
-    LT("lt", true),
-    LE("le", true),
-    GT("gt", true),
-    GE("ge", true);
+public enum Operator implements BiFunction<Predicate, EntityDictionary, java.util.function.Predicate> {
+    IN("in", true) {
+        @Override
+        java.util.function.Predicate getFilterFunction(String field, List<Object> values, EntityDictionary dictionary) {
```

Comment:
Instead of making these anonymous, could they be static functions that are just referenced in the enum: Operator::in